pinot-spi/src/main/java/org/apache/pinot/spi/stream/RowWithKey.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.spi.stream;
+
+import java.util.Arrays;
+import javax.annotation.Nullable;
+
+
+/**
+ * Helper class so the key and payload can be easily tied together instead of using pair
+ */
+public class RowWithKey {
```

Comment:
since this goes into SPI. we need to be careful explaining what's the intended usage. could you give a bit more in the javadoc?

At the first glance, I thought it is representing a row in a partitioned table with the partition key. 

Also this seems to be an interface used by pinot-tools, it might work better to put it in the pinot-tools module as a helper utils (along with produce batch)

Reply:
Good point. I moved the classed to be inside `StreamDataProducer` so it will not be used in other places. If in future people see the need for such class they can do refactor